pkg/mempool/custom_proposal_handler_test.go

```diff
@@ -0,0 +1,393 @@
+		{tx: buildMsg(s.T(), txConfig, []byte(`72`), [][]byte{secret3, secret5}, []uint64{4, 3}), priority: 8},
+		{tx: buildMsg(s.T(), txConfig, []byte(`82`), [][]byte{secret2, secret6}, []uint64{4, 3}), priority: 8},
+		// test 3
+		{tx: buildMsg(s.T(), txConfig, []byte(`9`), [][]byte{secret3, secret4}, []uint64{3, 3}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`1052345678910`), [][]byte{secret1, secret2}, []uint64{4, 4}), priority: 8},
+		{tx: buildMsg(s.T(), txConfig, []byte(`11`), [][]byte{secret1, secret2}, []uint64{5, 5}), priority: 8},
+		// test 4
+		{tx: buildMsg(s.T(), txConfig, []byte(`1252345678910`), [][]byte{secret1}, []uint64{3}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`13`), [][]byte{secret1}, []uint64{5}), priority: 10},
+		{tx: buildMsg(s.T(), txConfig, []byte(`14`), [][]byte{secret1}, []uint64{6}), priority: 8},
+		// test 5, (same as test 1, but ethermint txs)
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`0`), [][]byte{secret1}, []uint64{1}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`12345678910`), [][]byte{secret1}, []uint64{2}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`22`), [][]byte{secret1}, []uint64{3}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`32`), [][]byte{secret2}, []uint64{1}), priority: 8},
+		// test 6, (same as test 4, but ethermint txs)
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`1252345678910`), [][]byte{secret1}, []uint64{3}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`13`), [][]byte{secret1}, []uint64{5}), priority: 10},
+		{tx: buildEthMsg(s.T(), txConfig, []byte(`14`), [][]byte{secret1}, []uint64{6}), priority: 8},
+		// large cosmos tx to test with eth txs
+		{
+			tx: buildMsg(
+				s.T(),
+				txConfig,
+				[]byte(`1252345678910125234567891012523456789101252345678910125234567891012523456789101252345678910125234567891012523456789101252345678912343`),
+				[][]byte{secret2},
+				[]uint64{1},
+			),
+			priority: 8,
+		},
+	}
+
+	for i := range testTxs {
+		bz, err := txConfig.TxEncoder()(testTxs[i].tx)
+		s.Require().NoError(err)
+		testTxs[i].bz = bz
+		testTxs[i].size = int(cmttypes.ComputeProtoSizeForTxs([]cmttypes.Tx{bz}))
+	}
+
+	s.Require().Equal(testTxs[0].size, 111)
+	s.Require().Equal(testTxs[1].size, 121)
+	s.Require().Equal(testTxs[2].size, 112)
+	s.Require().Equal(testTxs[3].size, 112)
+	s.Require().Equal(testTxs[4].size, 195)
+	s.Require().Equal(testTxs[5].size, 205)
+	s.Require().Equal(testTxs[6].size, 196)
+	s.Require().Equal(testTxs[7].size, 196)
+	s.Require().Equal(testTxs[8].size, 196)
+
+	// ethermint txs
+	s.Require().Equal(testTxs[15].size, 247)
+	s.Require().Equal(testTxs[16].size, 257)
+	s.Require().Equal(testTxs[17].size, 248)
+	s.Require().Equal(testTxs[18].size, 248)
+
+	s.Require().Equal(testTxs[19].size, 259)
+	s.Require().Equal(testTxs[20].size, 248)
+	s.Require().Equal(testTxs[21].size, 248)
+
+	s.Require().Equal(testTxs[22].size, 248)
+
+	testCases := map[string]struct {
```

Comment:
just curious is there any reason for not using `[]struct` , and using `map[string]struct` 
Is the order of execution important for these tests? 

If not, we can consider using `[]struct` as we have been using that format for the rest of the table-driven tests . 
note : I don't see any issues with using the map